core/lib/src/local/client.rs

```diff
@@ -1,396 +1,238 @@
-    ///
-    /// When dispatched, the request will be served by the instance of Rocket
-    /// within `self`. The request is not dispatched automatically. To actually
-    /// dispatch the request, call [`LocalRequest::dispatch()`] on the returned
-    /// request.
-    ///
-    /// # Example
-    ///
-    /// ```rust
-    /// use rocket::local::Client;
-    ///
-    /// # rocket::async_test(async {
-    /// let client = Client::new(rocket::ignite()).await.expect("valid rocket");
-    /// let req = client.put("/hello");
-    /// # });
-    /// ```
-    #[inline(always)]
-    pub fn put<'c, 'u: 'c, U: Into<Cow<'u, str>>>(&'c self, uri: U) -> LocalRequest<'c> {
-        self.req(Method::Put, uri)
-    }
-
-    /// Create a local `POST` request to the URI `uri`.
-    ///
-    /// When dispatched, the request will be served by the instance of Rocket
-    /// within `self`. The request is not dispatched automatically. To actually
-    /// dispatch the request, call [`LocalRequest::dispatch()`] on the returned
-    /// request.
-    ///
-    /// # Example
-    ///
-    /// ```rust
-    /// use rocket::local::Client;
-    /// use rocket::http::ContentType;
-    ///
-    /// # rocket::async_test(async {
-    /// let client = Client::new(rocket::ignite()).await.expect("valid rocket");
-    ///
-    /// let req = client.post("/hello")
-    ///     .body("field=value&otherField=123")
-    ///     .header(ContentType::Form);
-    /// # });
-    /// ```
-    #[inline(always)]
-    pub fn post<'c, 'u: 'c, U: Into<Cow<'u, str>>>(&'c self, uri: U) -> LocalRequest<'c> {
-        self.req(Method::Post, uri)
+macro_rules! impl_client {
+    ($import:literal $(@$prefix:tt $suffix:tt)? $name:ident) =>
+{
+    impl $name {
+        /// Construct a new `Client` from an instance of `Rocket` with cookie
+        /// tracking.
+        ///
+        /// # Cookie Tracking
+        ///
+        /// By default, a `Client` propagates cookie changes made by responses
+        /// to previously dispatched requests. In other words, if a previously
+        /// dispatched request resulted in a response that adds a cookie, any
+        /// future requests will contain the new cookies. Similarly, cookies
+        /// removed by a response won't be propagated further.
+        ///
+        /// This is typically the desired mode of operation for a `Client` as it
+        /// removes the burden of manually tracking cookies. Under some
+        /// circumstances, however, disabling this tracking may be desired. The
+        /// [`untracked()`](Client::untracked()) method creates a `Client` that
+        /// _will not_ track cookies.
+        ///
+        /// # Errors
+        ///
+        /// If launching the `Rocket` instance would fail, excepting network errors,
+        /// the `LaunchError` is returned.
+        ///
+        /// ```rust,no_run
+        #[doc = $import]
+        ///
+        /// let rocket = rocket::ignite();
+        /// let client = Client::new(rocket);
```

Comment:
This line needs an `.await` in the asynchronous version. Is there a nice way to "paste" the `$(.$suffix)?` into the line instead of adding arguments like `$call_new:literal`? I can tell `#[doc = $(.$suffix)?]` won't work for several reasons. The same issue is present for the `async fn`s on `LocalRequest` and `LocalResponse`.